ocs_ci/utility/aws.py

```diff
@@ -462,5 +587,77 @@ def get_data_volumes(instance_id):
     # Get the data volume according to DeleteOnTermination
     return [
         vol for vol in volumes if vol.attachments[0]
-        .get('DeleteOnTermination') is False
+                                      .get('DeleteOnTermination') is False
     ]
+
+
+def get_vpc_id_by_node_obj(aws_obj, instances):
+    """
+    This function getting vpc id by randomly selecting instances out of user aws deployment
+
+    Args:
+        aws_obj (obj): AWS() object
+        instances (dict): cluster ec2 instances objects
+
+    Returns:
+        str: vpc_id: The vpc id
+
+    """
+
+    instance_id = random.choice(list(instances.keys()))
+    vpc_id = aws_obj.get_vpc_id_by_instance_id(instance_id)
+
+    return vpc_id
+
+
+def block_availability_zone(security_group_id, instances):
+    """
+    Block all instances in randomly selected aws availability zone(az)
+    - Fetch az out of cluster instances
+    - Append security group without access permissions to all cluster nodes within same az
+
+    Args:
+        security_group_id (str): security group without access permissions
+        instances (dict): cluster ec2 instances objects
+
+    Returns:
+        dict: keys: blocked instances: all instances within the same availability zone
+              values: list of original security groups
+
+    """
+    aws_obj = AWS()
+    instances_to_block, current_sgs, original_sgs = [], [], []
+    instances_id = list(instances.keys())
+    random_az_selector = random.choice(instances_id)
+    random_az_selected = aws_obj.get_availability_zone_id_by_instance_id(random_az_selector)
+    logger.info(f"random_az: {random_az_selector}")
+    for instance in instances_id:
+        az = aws_obj.get_availability_zone_id_by_instance_id(instance)
+        if random_az_selected == az:
+            instances_to_block.append(instance)
+
+    logger.info(f"instances to block: {instances_to_block}")
+    for instance in instances_to_block:
+        original_sgs.append(aws_obj.get_security_groups_by_instance_id(instance))
+        aws_obj.append_security_group(security_group_id, instance)
+
+    for instance in instances_to_block:
+        for sg_grp in original_sgs:
+            for sg in sg_grp:
+                aws_obj.remove_security_group(sg, instance)
+
+    return dict(zip(instances_to_block, original_sgs))
```

Comment:
why do you have to return a dict here? is there a direct relation between each item in instances_to_block and original_sgs?
It's better to return a tuple:

```python
return instances_to_block, original_sgs
```

Reply:
dictionary is needed here, to store original state, each ec2 instance(key) with its original security groups(values)

Reply:
IMO, it would be better if this function was there just to block. This means you'll have to create the security group from and pick the instances from outside and send them to the function, so it'll just append_security_group

Reply:
ok, i will split that into 2 methods

Reply:
in order to block instance, it is required to:
1. append new security group without permissions
2. remove any other security group
3. ec2 instance must have at least 1 security group

Reply:
done, plz review again